src/mmrelay/db_utils.py

```
@@ -229,70 +398,105 @@ def delete_plugin_data(plugin_name, meshtastic_id):
+            plugin_name,
+            meshtastic_id,
+        )
+        return []
+
+    try:
+        return json.loads(result[0] if result else "[]")
+    except (json.JSONDecodeError, TypeError) as e:
         logger.error(
-            f"Database error retrieving plugin data for {plugin_name}, node {meshtastic_id}: {e}"
+            f"Failed to decode JSON data for plugin {plugin_name}, node {meshtastic_id}: {e}"
         )
```

Ruff (0.14.3)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)